gc/base/MemoryPool.hpp

```diff
@@ -337,7 +337,15 @@ class MM_MemoryPool : public MM_BaseVirtual
 	 * Increase the dark matter estimate for the receiver by the specified amount
 	 * @param bytes the number of bytes to increase the recorded estimate
 	 */
-	MMINLINE void incrementDarkMatterBytes(uintptr_t bytes) { _darkMatterBytes += bytes; }
+	MMINLINE void incrementDarkMatterBytes(uintptr_t bytes, bool needSync = false)
+	{
+		if (!needSync) {
+			_darkMatterBytes += bytes;
+		} else {
+			MM_AtomicOperations::add(&_darkMatterBytes, bytes);
+		}
```

Comment:
Would you please remove "!" and swap halves of "if"?

		if (needSync) {
 			MM_AtomicOperations::add(&_darkMatterBytes, bytes);
 		} else {
 			_darkMatterBytes += bytes;
 		}

Reply:
I kind of don't like the runtime _if_ (we could just have 2 versions of this method), but I hope that compiler can resolve needSync at  all call sites and  optimizes out the runtime check.

Reply:
I updated with two versions of this method, incrementDarkMatterBytes and incrementDarkMatterBytesAtomic